[Task]: Fix github.com/tetratelabs/wazero from 1.0.0-pre.7 to 1.0.0-pre.8 in /sdks

### What needs to happen?

PR #25249 is failing because wazero made a breaking change to their API:
```
examples/wasm/wasm.go:118:20: too many arguments in call to fn.r.NewHostModuleBuilder("env").NewFunctionBuilder().WithFunc(logString).Export("log").Instantiate
 	have (context.Context, wazero.Runtime)
 	want (context.Context)
```

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [X] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [X] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/go.mod
+++ b/sdks/go.mod
@@ -48,7 +48,7 @@ require (
 	github.com/proullon/ramsql v0.0.0-20211120092837-c8d0a408b939
 	github.com/spf13/cobra v1.6.1
 	github.com/testcontainers/testcontainers-go v0.15.0
-	github.com/tetratelabs/wazero v1.0.0-pre.7
+	github.com/tetratelabs/wazero v1.0.0-pre.9
 	github.com/xitongsys/parquet-go v1.6.2
 	github.com/xitongsys/parquet-go-source v0.0.0-20220315005136-aec0fe3e777c
 	go.mongodb.org/mongo-driver v1.11.1

--- a/sdks/go/examples/wasm/README.md
+++ b/sdks/go/examples/wasm/README.md
@@ -33,7 +33,7 @@ BEAM_HOME=$(pwd)/beam
 cd $BEAM_HOME/sdks
 ```
 
-## Direct Runner execution
+## Local Runner execution
 
 To execute this example on the local runner:
 

--- a/sdks/go/examples/wasm/README.md
+++ b/sdks/go/examples/wasm/README.md
@@ -35,7 +35,7 @@ cd $BEAM_HOME/sdks
 
 ## Direct Runner execution
 
-To execute this example on the direct runner:
+To execute this example on the local runner:
 
 ```shell
 go run ./go/examples/wasm --output=$OUTPUT

--- a/sdks/go/examples/wasm/wasm.go
+++ b/sdks/go/examples/wasm/wasm.go
@@ -81,7 +81,7 @@ func main() {
 func run(ctx context.Context) error {
 	p, s := beam.NewPipelineWithRoot()
 
-	in := beam.Create(s, "Ada", "Lovelace", "World", "Beam", "Senior López")
+	in := beam.Create(s, "Ada", "Lovelace", "World", "Beam", "Senior López", "Random unicorn emoji 🦄")
 
 	out := beam.ParDo(s, &embeddedWasmFn{}, in)
 
@@ -115,14 +115,14 @@ func (fn *embeddedWasmFn) Setup(ctx context.Context) error {
 	// log to the console.
 	_, err := fn.r.NewHostModuleBuilder("env").
 		NewFunctionBuilder().WithFunc(logString).Export("log").
-		Instantiate(ctx, fn.r)
+		Instantiate(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to instantiate host module: %w", err)
 	}
 
 	// Instantiate a WebAssembly module that imports the "log" function defined
 	// in "env" and exports "memory" and functions we'll use in this example.
-	fn.mod, err = fn.r.InstantiateModuleFromBinary(ctx, greetWasm)
+	fn.mod, err = fn.r.Instantiate(ctx, greetWasm)
 	if err != nil {
 		return fmt.Errorf("failed to instantiate wasm module: %v", err)
 	}

